[circle-mlir/tools-test] Introduce initial rewrite-test (#14828)
Samsung/ONE: circle-mlir/circle-mlir/tools-test/onnx2circle-rewrite-test/README.md

## onnx2circle-rewrite-test

> This test is to validate "RewriteCirclePass" is working as expected.

Reviewer: I have refined the sentence a bit.

```suggestion
This test validates whether the "RewriteCirclePass" works as expected.
```

> Expectation is validated with expected operators should exist and non-expected
> operators should not exist.

Reviewer: Also I've refined it. 😅

```suggestion
The test checks:
- Expected operators exist in the output.
- Non-expected operators do not exist.
```
